pkg/core/consensus/reduction/firststep/reducer.go

```diff
@@ -127,47 +124,72 @@ func (r *Reducer) sendReduction(step uint8, hash []byte) {
 		lg.WithField("category", "BUG").WithError(err).Errorln("error in signing reduction")
 		return
 	}
+	red := message.NewReduction(hdr)
+	red.SignedHash = sig
+	msg := message.New(topics.Reduction, *red)
 
-	payload := new(bytes.Buffer)
-	if err := encoding.WriteBLS(payload, sig); err != nil {
-		lg.WithField("category", "BUG").WithError(err).Errorln("error in encoding BLS signature")
-		return
+	if err := r.signer.Gossip(msg, r.ID()); err != nil {
+		lg.WithField("category", "BUG").WithError(err).Errorln("error in sending authenticated Reduction")
 	}
+}
 
-	if err := r.signer.Gossip(topics.Reduction, hdr, payload, r.ID()); err != nil {
-		lg.WithField("category", "BUG").WithError(err).Errorln("error in sending authenticated Reduction")
+// StepVotesMsgFactory creates a StepVotesMsg to be passed internally within
+// the Consensus components
+type StepVotesMsgFactory struct {
+	sv   message.StepVotes
+	hash []byte
+}
+
+// Create a new StepVotesMsgFactory
+func (s StepVotesMsgFactory) Create(sender []byte, round uint64, step uint8) consensus.InternalPacket {
+	// handling the case if the StepVotes is empty
+	// TODO: interface - check this. The step sounds like it is intrinsic
```

Comment:
The step can essentially be left out of the StepVotes, if I'm not mistaken. With the consensus refactoring, steps progress in sequence between all of the components, and thus the reduction steps can be inferred. This is probably more appropraite in a separate issue though, and should be checked.

Reply:
this indeed needs to be addressed through a low-priority separate issue. Will open one